### PR TITLE
Add evidence locker TOC

### LINKED ISSUE #38: Provide Evidence locker TOC for reports
## Overview

We need a summary report that lists all of the current reports with descriptions and links to those reports within the evidence locker.

## Requirements

- Report will provide a markdown summary of all reports in the evidence locker.
- Report will serve as a table of contents for reports.
- Report will provide links to reports.
- Report will organize reports.
- Report will provide links to evidence used.

## Approach

- The plan is to put this report content directly into the evidence locker readme.

## Security and Privacy

N/A

## Test Plan

TBD

### PR DESCRIPTION
**Pull request by alfinkel**
- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)

## What

- Add evidence locker TOC
- Add helper method to ComplianceCheck for retrieving historical evidence

## Why

- TOC provides a nicely formatted table of contents for reports with links to reports, evidence used, and other details

## How

- Update ReportBuilder logic to include appending a TOC to the end of an evidence locker's README

## Test

- All fetchers and checks and reports all function as before
- TOC is rendered on a locker's README as expected

## Context

Closes #38 
Also ref: #70

### CODE REVIEW

File: compliance/locker.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
 INDEX_FILE = 'index.json'
 DAY = 60 * 60 * 24
 AE_DEFAULT = 30 * DAY
-AE_EXEMPT = [INDEX_FILE, 'README.md', 'readme.md']
+AE_EXEMPT = [INDEX_FILE, 'README.md', 'readme.md', 'Readme.md']
```
Comment (drsm79) [inline compliance/locker.py:44]: What about mkaing this case insensative vi `re`?

File: compliance/report.py [hunk 2]
```diff
@@ -99,27 +99,86 @@ def get_template_for(self, test_obj, evidence):
...
+    def generate_toc(self, rpt_metadata):
+        """
+        Generate a check reports table of contents.
+
+        This method generates a TOC based on all report evidence metadata and
+        appends that TOC to the bottom of an evidence locker's README.md file.
+
+        :param rpt_metadata: Metadata from all report evidence index.json files
+        """
+        readme = 'README.md'
```
Comment (drsm79) [inline compliance/report.py:111]: pathlib might be nice here...

**Review by drsm79 — COMMENTED** [review]
Can you point to an example TOC? A gist would be fine.

File: compliance/locker.py (hunk 1 above)
Comment (alfinkel) [reply compliance/locker.py:44]: I thought about that but I decided against it to guard against exempting files like 'rEaDmE.mD`.

File: compliance/report.py (hunk 2 above)
Comment (alfinkel) [reply compliance/report.py:111]: how so?

**Comment by alfinkel** [comment]
[Example TOC]((https://gist.github.com/alfinkel/538a4e4c4279ee659580f074f684045e)

File: compliance/report.py (hunk 2 above)
Comment (alfinkel) [reply compliance/report.py]: I replaced all usage of the `os` package with the `pathlib` package in https://github.com/ComplianceAsCode/auditree-framework/pull/71/commits/5c5a7e607bcb7340b919fd3588f42873e3970aff

**Review by drsm79 — APPROVED** [review]

**Pull request merged**